selfhost/main.jakt

```diff
@@ -1366,19 +1366,30 @@ struct Parser {
                 return_type = .parse_typename()
                 return_type_span = JaktSpan(start, end: .index)
             }
-            else => {
-                .error("Expected ->", .current().span())
-            }
+            else => {}
         }
 
-        // FIXME: Accept an (optional) fat arrow ('=>')
-
-        parsed_function.block = .parse_block()
+        
+        parsed_function.block = .parse_fat_arrow()
```

Comment:
This looks a bit strange. If there's no fat arrow present, the function block is still parsed by `parse_fat_arrow()`.
Please rearrange the code so that `parse_fat_arrow()` isn't responsible for the non-fat-arrow case. :)

Reply:
I was trying to avoid having two `match .current() {` right next to each other, but I will rearrange.